src/context/CartesianGraphicalItemContext.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { createContext, useCallback, useContext, useEffect } from 'react';
+import { ErrorBarsSettings } from '../state/graphicalItemsSlice';
+import { SetCartesianGraphicalItem } from '../state/SetCartesianGraphicalItem';
+import { ChartData } from '../state/chartDataSlice';
+import { AxisId } from '../state/axisMapSlice';
+import { DataKey } from '../util/types';
+
+const noop = () => {};
+
+type DispatchPayload = {
+  addErrorBar: (errorBar: ErrorBarsSettings) => void;
+  removeErrorBar: (errorBar: ErrorBarsSettings) => void;
+};
+
+const ErrorBarDirectionDispatchContext = createContext<DispatchPayload>({
+  addErrorBar: noop,
+  removeErrorBar: noop,
+});
+
+type GraphicalItemContextProps = {
+  data: ChartData;
+  xAxisId: AxisId;
+  yAxisId: AxisId;
+  dataKey: DataKey<any>;
+  children: React.ReactNode;
+};
+
+export const CartesianGraphicalItemContext = ({
```

Comment:
Why not redux for all of this?

Reply:
We need to keep track which error bar belongs to which graphical item because that's where the axis id is defined. It does end up in the redux store in the end, the context here is necessary to avoid yet another layer of direct Dom access when rendering the error bars.